apps/hes/fsw/src/hes_app.c

```diff
@@ -0,0 +1,1012 @@
+					break;
+				}
+
+				case TO_CPD(PX4_VEHICLE_LAND_DETECTED_MID):
+				{
+					HES_AppData.HkTlm.VLandDetectedMsgRcvCnt++;
+					CFE_PSP_MemCpy(&HES_AppData.CVT.VLandDetected, DataMsgPtr, sizeof(HES_AppData.CVT.VLandDetected));
+					break;
+				}
+
+				case TO_CPD(PX4_VEHICLE_STATUS_MID):
+				{
+					HES_AppData.HkTlm.VehicleStatusMsgRcvCnt++;
+					CFE_PSP_MemCpy(&HES_AppData.CVT.VehicleStatus, DataMsgPtr, sizeof(HES_AppData.CVT.VehicleStatus));
+					break;
+				}
+
+                default:
+                    (void) CFE_EVS_SendEvent(HES_MSGID_ERR_EID, CFE_EVS_ERROR,
+                                      "Recvd invalid data msgId (0x%04X)", (unsigned short)DataMsgId);
+                    break;
+            }
+        }
+        else if (iStatus == CFE_SB_NO_MESSAGE)
+        {
+            break;
+        }
+        else
+        {
+            (void) CFE_EVS_SendEvent(HES_PIPE_ERR_EID, CFE_EVS_ERROR,
+                  "Data pipe read error (0x%08X)", (unsigned int)iStatus);
+            HES_AppData.uiRunStatus = CFE_ES_APP_ERROR;
+            break;
+        }
+    }
+}
+
+/* Conjugate with inversed quaternion */
+/* Body to NED Frame*/
+void Quaternion_Conjugate(const float* v, const float* data, float* result)
+{
+    float q0q0 = data[0] * data[0];
+    float q1q1 = data[1] * data[1];
+    float q2q2 = data[2] * data[2];
+    float q3q3 = data[3] * data[3];
+    
+    result[0] = (v[0] * (
+            q0q0 + q1q1 - q2q2 - q3q3) +
+            v[1] * 2.0f * (data[1] * data[2] - data[0] * data[3]) +
+            v[2] * 2.0f * (data[0] * data[2] + data[1] * data[3]));
+
+    result[1] = (v[0] * 2.0f * (data[1] * data[2] + data[0] * data[3]) +
+            v[1] * (q0q0 - q1q1 + q2q2 - q3q3) +
+            v[2] * 2.0f * (data[2] * data[3] - data[0] * data[1]));
+
+    result[2] = (v[0] * 2.0f * (data[1] * data[3] - data[0] * data[2]) +
+            v[1] * 2.0f * (data[0] * data[1] + data[2] * data[3]) +
+            v[2] * (q0q0 - q1q1 - q2q2 + q3q3));
+}
+
+
+/* Conjugate with inversed quaternion */
+/* NED to Body Frame*/
+void Quaternion_ConjugateInversed(const float* v, const float* data, float* result)
+{
+    float q0q0 = data[0] * data[0];
+    float q1q1 = data[1] * data[1];
+    float q2q2 = data[2] * data[2];
+    float q3q3 = data[3] * data[3];
+    
+    result[0] = (v[0] * (
+            q0q0 + q1q1 - q2q2 - q3q3) +
+            v[1] * 2.0f * (data[1] * data[2] + data[0] * data[3]) +
+            v[2] * 2.0f * (data[1] * data[3] - data[0] * data[2]));
+
+    result[1] = (v[0] * 2.0f * (data[1] * data[2] - data[0] * data[3]) +
+            v[1] * (q0q0 - q1q1 + q2q2 - q3q3) +
+            v[2] * 2.0f * (data[2] * data[3] + data[0] * data[1]));
+
+    result[2] = (v[0] * 2.0f * (data[1] * data[3] + data[0] * data[2]) +
+            v[1] * 2.0f * (data[2] * data[3] - data[0] * data[1]) +
+            v[2] * (q0q0 - q1q1 - q2q2 + q3q3));
+}
+
+
+void Quaternion_RotationMatrix(float* data, float** R)
```

Comment:
We already have this function (Matrix3F3 Quaternion::RotationMatrix(void) const) defined in "apps/px4lib/fsw/src/math/Quaternion.cpp". It is fully unit tested. Use it instead